handler/src/main/java/io/netty/handler/flush/FlushConsolidationHandler.java

```diff
@@ -132,14 +141,32 @@ public void handlerRemoved(ChannelHandlerContext ctx) throws Exception {
     }
 
     private void resetReadAndFlushIfNeeded(ChannelHandlerContext ctx) {
-        readInprogess = false;
+        readInProgress = false;
         flushIfNeeded(ctx);
     }
 
     private void flushIfNeeded(ChannelHandlerContext ctx) {
         if (flushPendingCount > 0) {
+            if (nextScheduledFlush != null) {
+                nextScheduledFlush.cancel(false);
+            }
             flushPendingCount = 0;
             ctx.flush();
         }
     }
+
+    private void scheduleFlush(final ChannelHandlerContext ctx) {
+        // Run as soon as possible, but still yield to give a chance for additional writes to enqueue.
+        nextScheduledFlush = ctx.channel().eventLoop().schedule(flushTask, 0, TimeUnit.MILLISECONDS);
```

Comment:
@olim7t shouldn't this be:

```java
if (nextScheduleFlush == null) {
    nextScheduledFlush = ctx.channel().eventLoop().schedule(flushTask, 0, TimeUnit.MILLISECONDS);
}
```

Otherwise you will mist to cancel all scheduled flushes once `flushIfNeeded` is called. Beside this I think there is no need to schedule multiple if the one scheduled did not run yet.